### ISSUE TITLE
SqlException: Ambiguous column name 'CountryId'.  A column has been specified more than once in the order by list. Columns in the order by list must be unique. 

### ISSUE THREAD

**Issue by aneequrrehman** [dotnet/efcore#14619]
```
var userRoles = await _dbContext.UserRoles
                .Include(ur => ur.Company)
                .Where(ur => ur.UserId == userId)
                .Select(ur => new
                {
                    ur.CompanyId,
                    CountryId = ur.Company.CountryId,
                    CompanyName =  ur.Company.FullName,
                    ur.RoleId
                })
                .GroupBy(ur => new { ur.CompanyId, ur.CountryId, ur.CompanyName }, ur => new { ur.RoleId })
                .ToListAsync();
```

Of course, there're ways to write this query in a better way. But this query was working fine in the previous versions of the EF Core.

If I change its order to the following, it works:

```
var userRoles = (await _dbContext.UserRoles
                .Include(ur => ur.Company)
                .Where(ur => ur.UserId == userId)
                .Select(ur => new
                {
                    ur.CompanyId,
                    CountryId = ur.Company.CountryId,
                    CompanyName =  ur.Company.FullName,
                    ur.RoleId
                })
                .ToListAsync())
                .GroupBy(ur => new { ur.CompanyId, ur.CountryId, ur.CompanyName }, ur => new { ur.RoleId });
```

The version is `EFCore 2.2.1` in an ASP.NET Core 2.2 app.

**Comment by smitpatel** [comment]
Should be resolved by #12098

**Comment by smitpatel** [comment]
Not supported query #17068
